rust/src/writer/record_batch.rs

```diff
@@ -372,7 +372,8 @@ pub(crate) fn divide_by_partition_values(
         })
         .collect::<Result<Vec<_>, DeltaWriterError>>()?;
 
-    let partition_ranges = lexicographical_partition_ranges(sorted_partition_columns.as_slice())?;
```

Comment:
#[allow(warnings)]  for now as [:lexicographical_partition_ranges ](https://arrow.apache.org/rust/arrow/compute/fn.lexicographical_partition_ranges.html) is Deprecated

Reply:
@wjones127 / @rtyler  - Datafusion packages have been upgraded

Reply:
FYI another dedicated PR was merged to upgrade datafusion, so you'll need to address conflicts with that. Should be quick.